agent.go

```diff
@@ -665,6 +665,12 @@ func (ep *endpoint) addServiceInfoToCluster(sb *sandbox) error {
 }
 
 func (ep *endpoint) deleteServiceInfoFromCluster(sb *sandbox, method string) error {
+
+	if !ep.serviceEnabled {
```

Comment:
I don't see the serviceEnable flag being saved into the store in the Marshal of the endpoint, are we sure that does make sense then this value?